aiks/canvas.cc

```diff
@@ -254,14 +254,16 @@ void Canvas::Save(bool create_subpass) {
 
 void Canvas::DrawTextFrame(TextFrame text_frame,
                            std::shared_ptr<GlyphAtlas> atlas,
-                           Point position) {
+                           Point position,
+                           Paint paint) {
   if (!atlas || !atlas->IsValid()) {
     return;
   }
 
   auto text_contents = std::make_shared<TextContents>();
   text_contents->SetTextFrame(std::move(text_frame));
   text_contents->SetGlyphAtlas(std::move(atlas));
+  text_contents->SetColor(paint.color);
```

Comment:
Can we instead make the API on text_contents `SetPaint`?

Flutter supports drawing text with gradients. Originally it only supported flat colors, but this changed overtime and led to some slight complexity in parts of the codebase where we have to support both color and paint and ignore color if paint is specified.

Reply:
> Can we instead make the API on text_contents `SetPaint`?

Paint is an Aiks concept. OTOH, TextContents is in the the entity framework. Parts of the paint are selectively ignored depending on the the draw call the paint is used with. Depending on how and when we add support for the additional features, we can setup split the task of reverse engineering Skia and actually implementing the functionality in two separate frameworks. If possible, I'd like to not burden the Entity framework API with getting to decide what parts of paint it will (and won't) respect.

Reply:
SG. At this point it won't be a big deal anyway since a lot of the complexity I'm talking about ends up living in dart:ui anyway